File: src/main/java/com/ject/studytrip/trip/domain/factory/TripFactory.java
package com.ject.studytrip.trip.domain.factory;

import com.ject.studytrip.member.domain.model.Member;
import com.ject.studytrip.trip.domain.model.Trip;
import com.ject.studytrip.trip.domain.model.TripCategory;
import java.time.LocalDate;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class TripFactory {
    public static Trip create(
            Member member,
            String name,
            String memo,
            TripCategory category,
            LocalDate endDate,
            int totalStamps) {
        return Trip.of(member, name, memo, category, endDate, totalStamps);
    }
}

File: src/main/kotlin/com/ject/studytrip/trip/domain/factory/TripFactory.kt
package com.ject.studytrip.trip.domain.factory

import com.ject.studytrip.member.domain.model.Member
import com.ject.studytrip.trip.domain.model.Trip
import com.ject.studytrip.trip.domain.model.TripCategory
import java.time.LocalDate

object TripFactory {
    fun create(
        member: Member,
        name: String,
        memo: String?,
        category: TripCategory,
        endDate: LocalDate?,
        totalStamps: Int,
    ): Trip = Trip.of(member, name, memo, category, endDate, totalStamps)
}
